Correct and update align release guidance (#104)
GSA/automate.fedramp.gov: content/about/release.md

- Provide content and semantic compatibility tied to a specific version, allowing version-based data to be used by different tools.
- Minimize disruption for tool maintainers by indicating when a major change is compatibility breaking.

> [Semantic Versioning (SemVer)](https://semver.org/), which is short for semantic versioning, is used as a common practice to version software. Using SemVer, version numbers convey meaning around the degree of compatibility resulting from modifications made to the underlying code or tool.

Reviewer: with Semantic Versioning spelled out at the start of the sentence, the following phrase becomes redundant, "which is short for semantic versioning"

Author: ```suggestion
[Semantic Versioning (SemVer)](https://semver.org/) is used as a common practice to version software. Using SemVer, version numbers convey meaning around the degree of compatibility resulting from modifications made to the underlying code or tool.
```

A SemVer version is the combination of *MAJOR*.*MINOR*.*PATCH* as follows:

- Incrementing the **MAJOR** version indicates significant incompatible changes.
- Incrementing the **MINOR** version adds functionality in a backward-compatible manner.
- Incrementing the **PATCH** version indicates backward-compatible bug fixes.

{{<callout>}}

> In the past, FedRAMP used Spock versioning, per [Architectural Decision Record #2](https://github.com/GSA/fedramp-automation/blob/fcdd25e3177d4bf31178041648fdc74d610146e9/documents/adr/0002-git-release-version-strategy.md) and deprecated it in [ADR #10](https://github.com/aj-stein-gsa/fedramp-automation/blob/1f854385d1d75c4040d2e611b33909f13cb761c1/documents/adr/0010-semantic-versions-only.md). The last release that uses the deprecated release and version guidance is [FedRAMP Version 2.0.0 for OSCAL 1.0.4](https://github.com/GSA/fedramp-automation/releases/tag/fedramp-2.0.0-oscal-1.0.4).

Author: NOTE: once the GSA/fedramp-automation#866 is complete, this link can be updated for final approval and merge.

```suggestion
In the past, FedRAMP used Spock versioning, per [Architectural Decision Record #2](https://github.com/GSA/fedramp-automation/blob/fcdd25e3177d4bf31178041648fdc74d610146e9/documents/adr/0002-git-release-version-strategy.md) and deprecated it in [ADR #10](TBD). The last release that uses the deprecated release and version guidance is [FedRAMP Version 2.0.0 for OSCAL 1.0.4](https://github.com/GSA/fedramp-automation/releases/tag/fedramp-2.0.0-oscal-1.0.4).
```

{{</callout>}}

## Versioning Examples

Examples of FedRAMP using SemVer in a sequence:

- `3.0.0` (MAJOR: 3, MINOR: 0, PATCH : 0) indicates the MAJOR release 3.
- `3.0.1` (MAJOR: 3, MINOR: 0, PATCH : 1) indicates the first PATCH release for MAJOR release 3.
- `3.1.0` (MAJOR: 3, MINOR: 1, PATCH : 0) indicates the first MINOR release for MAJOR release 3.

{{<callout>}}

> The last version to use Spock versioning, per [ADR #10](https://github.com/aj-stein-gsa/fedramp-automation/blob/1f854385d1d75c4040d2e611b33909f13cb761c1/documents/adr/0010-semantic-versions-only.md), is [FedRAMP Version 2.0.0 for OSCAL 1.0.4 (`fedramp-2.0.0-oscal-1.0.4`)](https://github.com/GSA/fedramp-automation/releases/tag/fedramp-2.0.0-oscal-1.0.4). All subsequent releases will use version tag described above.

Author: NOTE: once the https://github.com/GSA/fedramp-automation/pull/866 is complete, this link can be updated for final approval and merge.

```suggestion
The last version to use Spock versioning, per [ADR #10](TBD), is [FedRAMP Version 2.0.0 for OSCAL 1.0.4 (`fedramp-2.0.0-oscal-1.0.4`)](https://github.com/GSA/fedramp-automation/releases/tag/fedramp-2.0.0-oscal-1.0.4). All subsequent releases will use version tag described above.
```

{{</callout>}}

> The following illustrates some drivers that will motivate specific version changes. OSCAL data within a release will use `oscal-version` in `metadata` to identify the minimally required version.

Reviewer: This phrasing sounds like we will use the oldest OSCAL version that will validate the OSCAL data within the release. I don't think that's quite what we intended. Don't we want to indicate the latest version of OSCAL against which the syntax was validated?

```suggestion
The following illustrates some drivers that will motivate specific version changes. OSCAL data within a release will use `oscal-version` in `metadata` to identify the latest version of OSCAL against which the content was validated.
```

Author: Minimally required was the intent, we want to support a minimal version and, per semantic versioning and allow anything that is not a breaking change (so everything until the first major digit changes). For now, we will pick a minimum of `1.1.2`, but if there _is_ an upstream `1.1.5` or `1.2.0`, we may likely stick with `1.1.2` until things change, so that is the minimally required version.

Does that make sense?

Author: So point being, what we test against may not we want publish as the minimally required, or I woulda accept this change.
